.editorconfig

```diff
@@ -0,0 +1,26 @@
+# EditorConfig helps developers define and maintain consistent
+# coding styles between different editors and IDEs
+# editorconfig.org
+
+root = true
+
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+
+[*.rs]
+indent_style = space
+indent_size = 4
+
+[*.toml]
+indent_style = space
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = false
```

Comment:
Do we need to preserve the whitespace? If yes, then a comment explaining why would be nice.

Reply:
Indeed we need to, because double whitespace at end of line denotes a line break in Markdown. I've added a comment, both here in /src/tools/clippy/.editorconfig